### ISSUE TITLE
Running a container based on this image fails early on some machines but not on others

### ISSUE THREAD

**Issue by JimFluke** [Unidata/thredds-docker#300]
This is a non-security issue:

- I am using the unidata/thredds-docker:5.4 downloaded from dockerhub on 3/1/2024
- On the machine where I see the problem we have: 
  - OS: AlmaLinux release 8.4 (Electric Cheetah)
  - Docker version 20.10.7, build f0df350
  - docker-compose version 1.23.2, build 1110ad01
- On the above machine I see this in the `docker logs`:
> Cannot find /usr/local/tomcat/bin/setclasspath.sh
> This file is needed to run this program
- But, on another machine everything works fine. Its specs are:
  - CentOS Linux release 7.9.2009 (Core)
  - Docker version 24.0.7, build afdd53b
  - docker-compose version 1.29.2, build 5becea4c

I know this does not include all of my project specifics. I can add those, but I'm hoping someone has seen this error before.

**Comment by julienchastang** [comment]
A few things:

- Are there any error messages that precede that error message that may give additional clues?
- I wonder if there are differences with respect to permissions assigned to the docker daemon and how it interacts with file system permissions that could account for the differences you are seeing between operating systems?
- Out of curiosity, what happens when you run `docker run -it tomcat:latest` in the different scenarios you have outlined?

**Comment by JimFluke** [comment]
- No, that is all I get for error messages. Nothing before or after that, and no thredds or tomcat log files get created. But the logs directories are also volume mounted, so maybe changing that is something to try.
- I can check on the docker permissions and ask about that. I don't know of any differences off hand.
- Running tomcat was a good check! It gives me exactly the same error on the Alma 8.4 machine, and seemed to work much better on the Centos 7.9 machine. I'll keep looking at differences, like the docker file permissions.

**Comment by JimFluke** [comment]
Actually, I do have some more information:
- The `setclasspath.sh` file and the `/usr/local/tomcat/bin` directory from the error message are not volume mounted, so they only exist in the container.
- I can log into the container on the Alma 8.4 machine while the health check is still in progress, and see the `~tomcat/` directory and its `bin/` subdirectory and everything is there. But, it is all owned by root, and it stays owned by root until the container aborts.

**Comment by ana-v-espinoza** [comment]
Hey Jim,

TL;DR:
Updating Docker helped somebody in a similar situation in the past.

---

After reviewing this issue, I had the feeling that we had, in fact, seen this problem before. Looking back at our closed issues I found #295 which describes the same thing you have.

What we noticed in that case and in yours is that your Docker version is out of date. In your case, your Alma Linux machine is on Docker v2.10.7.

In the issue referenced above, the issue was resolved by updating Docker to a more recent version. Unfortunately, we weren't able to determine the actual root cause of the problem, so if updating Docker is out of the question, I'm not quite sure what the next steps would be.

This seems to be more of a Docker issue than a `thredds-docker` issue, however, since you receive a similar error from the official `tomcat` image.

Nevertheless, I'll leave this issue open for a few more days in case something else comes up :)

Best,

Ana

**Comment by JimFluke** [comment]
Thanks Ana! I've been coming to the same conclusion. I've been working on a simple test container that, hopefully, will have the same issue. Then I can talk to our sys admin about it.

**Comment by JimFluke** [comment]
We upgraded Docker on our AlmaLinux 8.4 machine and it's all working fine now. So I will close this issue.
Thanks for all your help!
